objcgen/NameGenerator.cs

```diff
@@ -232,6 +232,8 @@ public static string GetExtendedParameterName (ParameterInfo p, ParameterInfo []
 		{
 			string pName = p.Name;
 			string ptname = GetTypeName (p.ParameterType);
+			if (RestrictedNames.IsRestrictedName (pName))
+				pName = "managed" + pName;
```

Comment:
it's not very consistent, i.e. the name change happens here, but the same situation will also happen outside parameter names, e.g. I can have an `auto` type, method, property...

It's also not very helpful to the final developer, it could even be confusing wrt CoreData.

It's been a while since I read the whole code... we might already have something similar but I think something like:

```
string pName = X.GetSafeName (p.Name);
```

is better since it centralize the logic, e.g. I'm sure we'll hit something else that's not a restricted name.

Also I'd rather use a `_` prefix but whatever we chose it must be verified against duplicates.

Reply:
I agree on managed being a less than ideal prefix. I stole it from the 1051 error cases, where we prefix with managed when it conflicts with "important" obj-c selectors.

I could factor out creating a safe name, but we're already in NameGenerator.cs so it's just be an extra method call inside there. 

This case seems "special" since its when we have a property and a method gen to the same name, neither with params for us to distinguish. Java just drops the get prefix on one fwict.